src/Symfony/Component/Console/Attribute/Input.php

```
@@ -46,46 +49,72 @@ public static function tryFrom(\ReflectionParameter|\ReflectionProperty $member)
         $self->class = new \ReflectionClass($class);
 
         foreach ($self->class->getProperties() as $property) {
-            if (!$property->isPublic() || $property->isStatic()) {
-                continue;
-            }
-
             if ($argument = Argument::tryFrom($property)) {
                 $self->definition[$property->name] = $argument;
-                continue;
-            }
-
-            if ($option = Option::tryFrom($property)) {
+            } elseif ($option = Option::tryFrom($property)) {
                 $self->definition[$property->name] = $option;
-                continue;
+            } elseif ($input = self::tryFrom($property)) {
+                $self->definition[$property->name] = $input;
             }
 
-            if ($input = self::tryFrom($property)) {
-                $self->definition[$property->name] = $input;
+            if (isset($self->definition[$property->name]) && (!$property->isPublic() || $property->isStatic())) {
+                throw new LogicException(\sprintf('The input property "%s::$%s" must be public and non-static.', $self->class->name, $property->name));
             }
         }
 
         if (!$self->definition) {
             throw new LogicException(\sprintf('The input class "%s" must have at least one argument or option.', $self->class->name));
         }
 
+        foreach ($self->class->getMethods() as $method) {
+            if ($self->interactive = Interactive::tryFrom($method)) {
+                break;
```

What if we put the `Interactive` attribute on multiple methods ? Right now, it seems to silently ignore any except the first one.

---

Given that the implementation relies on an iterable in `getMethodInteractions`, I think we could probably support any number of interactive methods just by using an array as storage and yielding from it.

---

It works for me as well. I've updated it so we allow any number of interactive methods, either in DTOs or in the invokable command class